contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -4503,6 +4514,45 @@ pltsql_validator(PG_FUNCTION_ARGS)
 		}
 
 		ReleaseSysCache(tuple);
+		
+		/* 
+		 * If the function has TVP in its arguments or function body 
+		 * it should be declared as VOLATILE by default 
+		 */
+		if(prokind == PROKIND_FUNCTION && (has_table_var && !is_itvf && !is_mstvf))
+		{
```

Comment:
Why are we doing this in validator

Reply:
So the variables related to table variables are set inside `do_compile` function. Specially inside `antlr_parser_cpp(proc_source)`. So actually the function is already created by that time. So I am updating the tuple of the function.

bt when a function is created 
```
do_compile
pltsql_compile
pltsql_validator
fmgr_security_definer 
FunctionCall1Coll
OidFunctionCall1Coll
ProcedureCreate
CreateFunction 
bbf_ProcessUtility
```

The main reason I use do changes inside `pltsql_validator` because we have access to relevant variable for table variable here, This function attempts to validate a PL/tsql function at CREATE FUNCTION time and no engine changes required.

Reply:
Is it possible to do it in post_object_create hook? I am assuming it won't because it will be executed directly after CreateFunction/ProcedureCreate but it'll be better to verify.

Reply:
It won't be possible to do it in post_object_create as we cannot get relevant variable get presence of tvp.